tests/integration/test_split.py

```diff
@@ -0,0 +1,310 @@
+        with pytest.raises(expected_exc):
+            num.vsplit(ary, 1)
+            # cuNumeric returns the array
+        with pytest.raises(expected_exc):
+            np.vsplit(ary, 1)
+            # Numpy raises
+            # ValueError: vsplit only works on arrays of 2 or more dimensions
+
+    @pytest.mark.xfail
+    def test_dimensions_hsplit_0(self):
+        ary = np.random.randint(low=0, high=100, size=(0,))
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.hsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.hsplit(ary, 1)
+            # Numpy returns  [array([], dtype=int64)]
+
+    def test_dimensions_hsplit_1(self):
+        ary = num.arange(10)
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.hsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.hsplit(ary, 1)
+
+    @pytest.mark.parametrize("size", ((0,), (10,), (5, 5)))
+    def test_dimensions_dsplit(self, size):
+        ary = np.random.randint(low=0, high=100, size=size)
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.dsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.dsplit(ary, 1)
+
+
+def compare_array(a, b):
+    """
+    Compare two array using zip method.
+    """
+    if len(a) != len(b):
+        return False
+    else:
+        for each in zip(a, b):
+            if not np.array_equal(*each):
+                return False
+    return True
+
+
+def get_indices(arr, axis):
+    indices_arr = []
+
+    if arr.shape[axis] == 1:
+        indices_arr.append(1)  # in index
+
+    elif arr.shape[axis] > 1:
+        for div in range(2, (int)(math.sqrt(arr.shape[axis]) + 1)):
+            if arr.shape[axis] % div == 0:
+                indices_arr.append(div)
+
+                # in index
+                indices_arr.append(list(range(1, arr.shape[axis], div)))
+
+                # out index
+                indices_arr.append(
+                    list(
+                        range(
+                            0,
+                            arr.shape[axis] + div * np.random.randint(1, 10),
+                            div,
+                        )
+                    )
+                )
+                indices_arr.append(
+                    list(
+                        range(
+                            arr.shape[axis] + div * np.random.randint(1, 10),
+                            0,
+                            -div,
+                        )
+                    )
+                )
+
+        # if indivisible
+        if len(indices_arr) == 0:
+            indices_arr.append(1)
+            indices_arr.append(arr.shape[axis])
+            indices_arr.append(list(range(1, arr.shape[axis], 1)))
+
+            # out index
+            indices_arr.append(
+                list(
+                    range(0, arr.shape[axis] + 1 * np.random.randint(1, 10), 1)
```

Comment:
This seems identical to 

https://github.com/nv-legate/cunumeric/pull/813/files#diff-22094b84475c2ba7bcfc065a3beb81ed400b54d297054a38dce2008814a1d70dR230

with `div = 1`  

Perhaps there is some futher simplification/refactoring that can be done to reduce code near-duplication? Aso the value `arr.shape[axis]` is referenced in many places perhaps it can be assigned to a name once at the top. (When `black` starts spitting out multi-lined, multi-indented output like this I generally take that as a sign that something is too complicated)

Reply:
fix